### PR TITLE
#169258618  Sequelize setup

### PR DESCRIPTION
**Pull request by hezronkimutai**
#### What does this PR do?
- This PR sets up sequelize

#### Description of Task to be completed
- [X] Install sequelize `npm install sequelize sequelize-cli`
- [X] Install pg and pg h-store `npm install pg pg-hstore`
- [x] Update the package.json file.
- [X] Initialize sequelize `npm db-init`
- [X] Create a dummy`npm dummy-model`
- [X] Do migrations `npm db-migrate`
- [x] Write the descriptions in the README.md

#### How should this be manually tested?
- git clone `https://github.com/andela/team-odd-bn-backend.git`
- git checkout `ch-setup-sequelize-169258618`
- Fill the .env file with your details in the format in tthe .env.example file
- Do `npm install` to install dependencies.
- Do `npm run db-migrate` to make migrations.
- Go to the psql cli to confirm that  the table have been created 

#### Any background context you want to provide?
 I have already created a dummy model, However if you want to create another model of your choice =, go to package.json file in the scripts find "dummy model", you can edit that script or add another script.
#### What are the relevant pivotal tracker stories?
- [#169258618](https://www.pivotaltracker.com/n/projects/2407428/stories/169258618)
#### Screenshots (if appropriate)
Screen Shot 2019-11-05 at 09 56 43

### CODE REVIEW

**Review by victkarangwa — COMMENTED** [review]
Good work @hezronkimutai 
However, I'm afraid we shouldn't have placed `src/database/config/config.js` on the repo for the sake of security risks of the DB.

File: package.json [hunk 1]
```diff
@@ -4,6 +4,10 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
+    "start": "NODE_ENV=development nodemon src/index.js",
+    "db-init": "sequelize init",
+    "dummy-model":"npx sequelize-cli model:generate --name User --attributes firstName:string,lastName:string,email:string",
```
Comment (NiyoEric) [inline package.json]: @hezronkimutai by default Postgres create table in lowercase `user` why you change it to plural case.

Comment (hezronkimutai) [reply package.json]: Okey, Let me change that.

File: .env.example [hunk 2]
```diff
@@ -0,0 +1 @@
+DATABASE_URL=your_database_url
```
Comment (julietezekwe) [inline .env.example]: End of line

File: src/database/config/config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  },
+
+  "production": {
+    "use_env_variable": "DATABASE_URL",
+    "dialect": "postgresql",
+    "operatorsAliases": false
+  }
+}
+
+module.exports = config
```
Comment (julietezekwe) [inline src/database/config/config.js]: End of Line

File: src/database/migrations/20191105091614-create-user.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.dropTable('users');
+  }
+};
```
Comment (julietezekwe) [inline src/database/migrations/20191105091614-create-user.js]: End of line

File: src/database/models/index.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const Sequelize = require("sequelize");
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || "development";
+const config = require(__dirname + "/../config/config.js")[env];
+const dotenv = require("dotenv");
+const db = {};
```
[the remark this answers is not in the document]
Comment (julietezekwe) [reply src/database/models/index.js:9]: Resolve Hound issue

**Review by julietezekwe — CHANGES_REQUESTED** [review]
Kindly fix all hound issues. And rebase from develop

**Review by nziokaivy — DISMISSED** [review]

**Review by nziokaivy — APPROVED** [review]

**Review by victkarangwa — APPROVED** [review]

**Review by NiyoEric — DISMISSED** [review]

**Review by NiyoEric — COMMENTED** [review]
@hezronkimutai Would you remove MongoDB comment in `src/index.js`  and everything related.

**Review by NiyoEric — APPROVED** [review]

**Review by WillyBoris — APPROVED** [review]

File: .babelrc [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+    "presets": [
+        "@babel/preset-env"
+    ]
+}
```
Comment (julietezekwe) [inline .babelrc]: End of line

File: .eslintignore [hunk 7]
```diff
@@ -0,0 +1 @@
+src/database/
```
Comment (julietezekwe) [inline .eslintignore]: End of line

**Review by julietezekwe — COMMENTED** [review]
LGTM! Please add all the end of lines and let me know so I can merge.

**Comment by julietezekwe** [comment]
Please make the commits one

File: .babelrc (hunk 6 above)
Comment (hezronkimutai) [reply .babelrc]: okey

File: .eslintignore (hunk 7 above)
Comment (hezronkimutai) [reply .eslintignore]: I have solved the end of line issue.

**Pull request closed without merging**